### PR TITLE
Included dependabot for gomod

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Included dependabot for gomod.
- https://github.com/envoyproxy/envoy/pull/20930#issuecomment-1106126126

Signed-off-by: naveensrinivasan <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

**Comment by naveensrinivasan** [comment]
@phlax 👀

**Comment by daixiang0** [comment]
I still prefer we should pin dependence version for all examples, no need to update daily.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -60,3 +60,18 @@ updates:
   directory: "/"
   schedule:
     interval: daily
+
+- package-ecosystem: "gomod"
+  directory: "/examples/grpc-bridge/server"
+  schedule:
+    interval: daily
+
+- package-ecosystem: "gomod"
```
Comment (phlax) [inline .github/dependabot.yml:69]: this seems to be duplicated

also i think there are others afaict that we should include:

```console
$ find . -name go.mod
./examples/ext_authz/auth/grpc-service/go.mod
./examples/load-reporting-service/go.mod
./examples/grpc-bridge/server/kv/go.mod
./examples/grpc-bridge/server/go.mod

```

**Review by phlax — CHANGES_REQUESTED** [review]

**Review by phlax — APPROVED** [review]
lgtm, thanks @naveensrinivasan

**Pull request merged**